astropy/coordinates/tests/test_representation.py

```diff
@@ -330,6 +322,62 @@ def test_representation_shortcuts(self):
             sph, UnitSphericalRepresentation, UnitSphericalDifferential)
         assert representation_equal_up_to_angular_type(got, expected)
 
+    def test_transform(self):
+
+        s1 = SphericalRepresentation(lon=[1, 2] * u.deg, lat=[3, 4] * u.deg,
+                                     distance=[5, 6] * u.kpc)
+
+        matrix = rotation_matrix(-10, "z", u.deg)
+
+        s2 = s1.transform(matrix)
+
+        assert_allclose(s2.lon.deg, s1.lon.deg + 10)
```

Comment:
In principle, here and below could just use `assert_allclose_quantity(s2.lon, s1.log+10*u.deg)`, which arguably is a bit clearer, but am not going to insist.

Reply:
On it!

Reply:
done.

Reply:
Thanks @mhvk  for the **many** reviews! This final direction is definitely A LOT better than the kludge code I initially pushed.